ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java

```diff
@@ -100,6 +129,82 @@ public UnresolvedExpression visitEvalClause(EvalClauseContext ctx) {
     return new Let((Field) visit(ctx.fieldExpression()), visit(ctx.expression()));
   }
 
+  /**
+   * Kmeans arguments.
+   */
+  @Override
+  public UnresolvedExpression visitCentroids(CentroidsContext ctx) {
+    return new Argument(CENTROIDS, getArgumentValue(ctx.centroids));
+  }
+
+  @Override
+  public UnresolvedExpression visitIterations(IterationsContext ctx) {
+    return new Argument(ITERATIONS, getArgumentValue(ctx.iterations));
+  }
+
+  @Override
+  public UnresolvedExpression visitDistance_type(Distance_typeContext ctx) {
```

Comment:
method name `visitDistance_type` is generated by antlr because the parameter name is `distance_type`. I agree we need to follow camelCase convention for java method, however, in this case, we want to keep command parameter names in consistency first.

Reply:
I figured out why it's like this. Here you're visiting each argument name defined in Lexer.g4. This may be unnecessary and may explode the AST builder. I think you should refer to https://github.com/opensearch-project/sql/blob/f186d04e37be701efdca83739e9cf637a4470cb6/ppl/src/main/antlr/OpenSearchPPLParser.g4#L184 to add a KmeanArg rule in Parser.g4 and only visit it here.

Reply:
I'm not sure if I understand you correctly, but I do want to visit each argument name defined in Lexer.g4 (so that I can parse them into a map and use it to build the request later). I used the label-visitor trick @penghuo suggested before, and it's been used on other commands like [stat](https://github.com/opensearch-project/sql/blob/main/ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java#L190) and [field](https://github.com/opensearch-project/sql/blob/main/ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java#L169). Just their labels are in camelCase so the visitor method name is also in camelCase. I can change the label name to camelCase if that's your concern. But the label name won't stay consistent with the parameter name anymore.

Reply:
+1 @dai-chen 's comments, we could avoid access each token. for example, https://github.com/opensearch-project/sql/blob/main/ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java#L339

Reply:
Synced with dai-chen offline and tried the `ctx.getChild(0)` and `ctx.getChild(2)` approach. But it seems like Antlr doesn't generate command context by columns, instead, it's generated by rows. So I don't think this approach will work. Gonna sync with dai-chen offline again.